pipelines/ligtheweight-component-and-container-operations/ligtheweight-component-and-container-operations.ipynb

```diff
@@ -0,0 +1,233 @@
+   "metadata": {},
+   "source": [
+    "### Prerequisites: Install or update the pipelines SDK\n",
+    "\n",
+    "You may need to **restart your notebook kernel** after updating the KFP sdk.\n",
+    "\n",
+    "This notebook is intended to be run from a Kubeflow notebook server. (From other environments, you would need to pass different arguments to the `kfp.Client` constructor.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# You may need to restart your notebook kernel after updating \n",
+    "!python3 -m pip install kfp-server-api --upgrade --user\n",
+    "!python3 -m pip install kfp --upgrade --user"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "EXPERIMENT_NAME = 'Combining pythong lightweigt components and container operations'        # Name of the experiment in the UI\n",
+    "BASE_IMAGE = 'tensorflow/tensorflow:2.0.0b0-py3'    # Base image used for components in the pipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp\n",
+    "import kfp.dsl as dsl\n",
+    "from kfp import compiler\n",
+    "from kfp import components\n",
+    "from kfp import gcp\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create pipeline component"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create a python function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dsl.python_component(\n",
+    "    name='add_op',\n",
+    "    description='adds two numbers',\n",
+    "    base_image=BASE_IMAGE  # you can define the base image here, or when you build in the next step. \n",
+    ")\n",
+    "def add(a: float, b: float) -> float:\n",
+    "    '''Calculates sum of two arguments'''\n",
+    "    print(a, '+', b, '=', a + b)\n",
+    "    return a + b\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Build a pipeline component from the function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Convert the function to a pipeline operation.\n",
+    "add_op = components.func_to_container_op(\n",
+    "    add,\n",
+    "    base_image=BASE_IMAGE, \n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create a container operations"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def echo_op(text1, text2):\n",
+    "    return dsl.ContainerOp(\n",
```

Comment:
We'd like out users to create reusable components in stead of just creating ContainerOp objects.
Creating component.yaml files is just as easy, but you're getting a real component that can be used somewhere else.